### ISSUE TITLE
Add EfficientDet model

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#951]
**Short Description**

EfficientDet is a type of **object detection** model, which utilizes several optimization and backbone tweaks, such as the use of a [BiFPN](https://paperswithcode.com/method/bifpn), and a compound scaling method that uniformly scales the resolution,depth and width for all backbones, feature networks and box/class prediction networks at the same time.

**Papers**

- [EfficientDet: Scalable and Efficient Object Detection](https://arxiv.org/abs/1911.09070)
- Cited by **2417** (until now)

**Existing Implementations**

- [Official: TF 2 (Keras)](https://github.com/google/automl/tree/master/efficientdet)

**Other Information**
- TF Model Garden: [Checkpoints](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md). 
- Unofficial: keras : https://github.com/xuannianz/EfficientDet
- [second most popular in kaggle env after yolo-v5 ( due to licence 📄 issue of v5  )

**added more**

- https://github.com/google/automl/blob/master/efficientdet/tf2/efficientdet_keras.py
- https://github.com/leondgarse/keras_cv_attention_models/tree/main/keras_cv_attention_models/efficientdet
- https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch

cc. @fsx950223 @mingxingtan @kartik4949 @xuannianz

**Comment by ianstenbit** [comment]
@LukeWood @tanzhenyu is this of interest as part of our current OD efforts?

**Comment by tanzhenyu** [comment]
Sounds good to me. @mingxingtan wdyt?

**Comment by shivance** [comment]
@tanzhenyu is this issue still welcoming contributions? If yes, then can I contribute ?

**Comment by ianstenbit** [comment]
Yes this issue is still welcoming contributions -- feel free to open a PR!

**Comment by innat-asj** [comment]
@srikesh-07 this may help to get started https://github.com/google/automl/blob/master/efficientdet/tf2/efficientdet_keras.py

**Comment by ianstenbit** [comment]
I see that both @shivance and @srikesh-07 have expressed interest in this -- I'll assign you both and you two can decide amongst yourselves how you'd like to move forward.

**Comment by ianstenbit** [comment]
Some thoughts about EfficientDet:

Our goal should be to offer
- and EfficientDET detector with swappable backbones
- a BiFPN implementation that can be swapped into RetinaNet or YOLOV8 or any other detection model that works with an FPN.

**Comment by srikesh-07** [comment]
Thanks for thoughts, Working on it.

**Comment by innat** [comment]
@srikesh-07 @shivance 
Here is another [implementation](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch), in Pytorch. It might be a valuable reference too. 

**official**

![img_inferred_d0_official](https://github.com/keras-team/keras-cv/assets/17668390/15f5887f-34f1-4ec9-ac53-d71e8006f884) 


[implement in torch](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch), 

![img_inferred_d0_this_repo](https://github.com/keras-team/keras-cv/assets/17668390/32f7949c-c037-4e74-b8f8-c44a40133a25)

**Comment by Tachi107** [comment]
Why was this closed? It seems to me that EfficientDet is still unavailable in KerasCV.

**Comment by innat** [comment]
@Tachi107 
Are you interested in contributing?

**Comment by Tachi107** [comment]
Il giorno ven 5 apr 2024 alle 23:55:49 -07:00:00, Mohammed Innat 
***@***.***> ha scritto:
> Are you interested in contributing?
> 

Hi Mohammed, I'd love to, but unfortunately I still don't have the 
skill set needed to do so. I'm still figuring out how this all works!

So while I really want to see EfficientDet - and EfficientDet-Lite 
especially, since pretty much all "official" implementations are either 
abandoned or deprecated - in KerasCV, I won't contribute to this myself.

**Comment by frostbyte012** [comment]
@innat is this still open?

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

**Comment by innat** [comment]
@sachinprasadhs 
> If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

Firstly, identifying which feature is required or missing can be done effectively, either by practitioners or the Keras team. Tickets in `keras-cv` were created only when such features were **NOT** available or when a specific need was encountered. Commenting after a significant delay and requesting the recreation of the same issue is neither efficient nor appropriate. Kindly move this ticket (and any other similar tickets with the same context) from `keras-cv` to `keras-hub`.

**Comment by innat** [comment]
> This issue is stale because it has been open for 14 days with no activity. It will be closed if no further activity occurs. Thank you.

GTH.
